pkg/tuned/controller.go

```diff
@@ -539,23 +669,38 @@ func writeOpenShiftTunedImageEnv() error {
 	return nil
 }
 
-func TunedRecommendFileWrite(profileName string) error {
-	klog.V(2).Infof("tunedRecommendFileWrite(): %s", profileName)
-	if err := os.MkdirAll(tunedRecommendDir, os.ModePerm); err != nil {
-		return fmt.Errorf("failed to create directory %q: %v", tunedRecommendDir, err)
-	}
-	f, err := os.Create(tunedRecommendFile)
-	if err != nil {
-		return fmt.Errorf("failed to create file %q: %v", tunedRecommendFile, err)
+func TunedRecommendFileWritePath(rfPath, profileName string) error {
+	rfDir := filepath.Dir(rfPath)
+	klog.V(2).Infof("tunedRecommendFileWrite(): %s %s", profileName, rfDir)
+	if err := os.MkdirAll(rfDir, os.ModePerm); err != nil {
+		return fmt.Errorf("failed to create directory %q: %v", rfDir, err)
 	}
-	defer f.Close()
-	if _, err = f.WriteString(fmt.Sprintf("[%s]\n", profileName)); err != nil {
-		return fmt.Errorf("failed to write file %q: %v", tunedRecommendFile, err)
+	data := []byte(fmt.Sprintf("[%s]\n", profileName))
+	if err := os.WriteFile(rfPath, data, 0644); err != nil {
+		return fmt.Errorf("failed to write file %q: %v", rfPath, err)
 	}
-	klog.Infof("written %q to set TuneD profile %s", tunedRecommendFile, profileName)
+	klog.Infof("tunedRecommendFileWrite(): written %q to set TuneD profile %s", rfPath, profileName)
 	return nil
 }
 
+func TunedRecommendFileReadPath(rfPath string) (string, error) {
```

Comment:
We need to read back the recommended file when we are in the recovering flow, when the operand start and there's no profile object available. Or, more truthfully, I haven't find a cleaner and more clever way to avoid reading back the recommended file.

Over time I grew to kinda like it because the code _wants_ to reconstruct the state from the file actually unpacked earlier on the node, but I'm open to alternatives.